### PR TITLE
feat(chart): remove mirrored openshift image from Longhorn

### PR DESCRIPTION
**Pull request by derekbit**
Signed-off-by: Derek Su <derek.su@suse.com>#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#9597

### CODE REVIEW

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just one feedback. Need Jilliam to review the comments in values.yaml. 

As discussed, need to update the readme and doc for helm values change and the openshift section in longhorn doc.

File: chart/questions.yaml [hunk 1]
```diff
@@ -167,14 +167,14 @@ questions:
     label: Longhorn CSI Liveness Probe Image Tag
     group: "Longhorn CSI Driver Images"
   - variable: image.openshift.oauthProxy.repository
-    default: longhornio/openshift-origin-oauth-proxy
```
[the remark this answers is not in the document]
Comment (derekbit) [reply chart/questions.yaml:170]: Removed. Thank you.

**Comment by derekbit** [comment]
> LGTM, just one feedback. Need Jilliam to review the comments in values.yaml.
> 
> As discussed, need to update the readme and doc for helm values change and the openshift section in longhorn doc.

Yes, will update the doc after @jillian-maroket reviews the doc of the PR.

File: chart/README.md [hunk 2]
```diff
@@ -119,8 +119,8 @@ The `values.yaml` contains items used to tweak a deployment of this chart.
 | image.longhorn.supportBundleKit.tag | string | `"v0.0.43"` | Tag for the Longhorn Support Bundle Manager image. |
 | image.longhorn.ui.repository | string | `"longhornio/longhorn-ui"` | Repository for the Longhorn UI image. |
 | image.longhorn.ui.tag | string | `"master-head"` | Tag for the Longhorn UI image. |
-| image.openshift.oauthProxy.repository | string | `"longhornio/openshift-origin-oauth-proxy"` | Repository for the OAuth Proxy image. This setting applies only to OpenShift users. |
-| image.openshift.oauthProxy.tag | float | `4.15` | Tag for the OAuth Proxy image. This setting applies only to OpenShift users. Specify OCP/OKD version 4.1 or later. The latest stable version is 4.15. |
+| image.openshift.oauthProxy.repository | string | `""` | Repository for the OAuth Proxy image. Specify the upstream image, for example, quay.io/openshift/origin-oauth-proxy. This setting applies only to OpenShift users. |
```
Comment (jillian-maroket) [inline chart/README.md:122]:
```suggestion
| image.openshift.oauthProxy.repository | string | `""` | Repository for the OAuth Proxy image. Specify the upstream image (for example, "quay.io/openshift/origin-oauth-proxy"). This setting applies only to OpenShift users. |
```

File: chart/values.yaml [hunk 3]
```diff
@@ -103,10 +103,10 @@ image:
       tag: v2.14.0
   openshift:
     oauthProxy:
-      # -- Repository for the OAuth Proxy image. This setting applies only to OpenShift users.
-      repository: longhornio/openshift-origin-oauth-proxy
-      # -- Tag for the OAuth Proxy image. This setting applies only to OpenShift users. Specify OCP/OKD version 4.1 or later. The latest stable version is 4.15.
-      tag: 4.15
+      # -- Repository for the OAuth Proxy image. Specify the upstream image, for example, quay.io/openshift/origin-oauth-proxy. This setting applies only to OpenShift users.
```
Comment (jillian-maroket) [inline chart/values.yaml:106]:
```suggestion
      # -- Repository for the OAuth Proxy image. Specify the upstream image (for example, "quay.io/openshift/origin-oauth-proxy"). This setting applies only to OpenShift users.
```

File: chart/README.md [hunk 4]
```diff
@@ -119,8 +119,8 @@ The `values.yaml` contains items used to tweak a deployment of this chart.
...
+| image.openshift.oauthProxy.tag | float | `""` | Tag for the OAuth Proxy image. Specify OCP/OKD version 4.1 or later. The Longhorn team has validated version 4.15, available at quay.io/openshift/origin-oauth-proxy:4.15. This setting applies only to OpenShift users. |
```
Comment (jillian-maroket) [inline chart/README.md:123]:
```suggestion
| image.openshift.oauthProxy.tag | float | `""` | Tag for the OAuth Proxy image. Specify OCP/OKD version 4.1 or later (including version 4.15, which is available at quay.io/openshift/origin-oauth-proxy:4.15). This setting applies only to OpenShift users. |
```

File: chart/values.yaml [hunk 5]
```diff
@@ -103,10 +103,10 @@ image:
...
+      repository: ""
+      # -- Tag for the OAuth Proxy image. Specify OCP/OKD version 4.1 or later. The Longhorn team has validated version 4.15, available at quay.io/openshift/origin-oauth-proxy:4.15. This setting applies only to OpenShift users.
```
Comment (jillian-maroket) [inline chart/values.yaml:108]:
```suggestion
      # -- Tag for the OAuth Proxy image. Specify OCP/OKD version 4.1 or later (including version 4.15, which is available at quay.io/openshift/origin-oauth-proxy:4.15). This setting applies only to OpenShift users.
```

**Review by jillian-maroket — COMMENTED** [review]
Review done. Just a few nits

**Review by jillian-maroket — APPROVED** [review]

File: deploy/longhorn-okd.yaml [hunk 6]
```diff
@@ -5115,7 +5115,7 @@ spec:
               topologyKey: kubernetes.io/hostname
       containers:
       - name: oauth-proxy
-        image: longhornio/openshift-origin-oauth-proxy:4.15
```
Comment (innobead) [inline deploy/longhorn-okd.yaml:5118]: If applying this manifest, users need to remember to add image and tag. Probably brief this in the doc.

**Review by innobead — APPROVED** [review]

File: deploy/longhorn-okd.yaml (hunk 6 above)
Comment (derekbit) [reply deploy/longhorn-okd.yaml:5118]: Added in the website PR.

**Comment by derekbit** [comment]
@mergify backport v1.7.x v1.6.x

**Pull request merged**